vmware-base/src/com/cloud/hypervisor/vmware/util/VmwareHelper.java

```diff
@@ -210,16 +208,13 @@ public static VirtualDevice prepareDvNicDevice(VirtualMachineMO vmMo, ManagedObj
         dvPortConnection.setPortgroupKey(morNetwork.getValue());
         dvPortBacking.setPort(dvPortConnection);
         nic.setBacking(dvPortBacking);
-        nic.setKey(30);
 
         connectInfo.setAllowGuestControl(true);
-        connectInfo.setConnected(conntected);
+        connectInfo.setConnected(connected);
         connectInfo.setStartConnected(connectOnStart);
         nic.setAddressType("Manual");
         nic.setConnectable(connectInfo);
         nic.setMacAddress(macAddress);
-
-        nic.setUnitNumber(deviceNumber);
         nic.setKey(-contextNumber);
```

Comment:
@rhtyd I know this is not your code, but there is something that called my attention (maybe you know why this is the way it is). At first I thought it was some dusty on my laptop screen... When the method `nic.setKey` is called, the code does `-contextNumber`, converting the positive integer to a negative one. I looked at the docs and the key is a unique ID of the NIC, but I reading the docs I did not see any reasons to use negative value as IDs.

Reply:
@rafaelweingartner when I was trying to fix the bug, I thought too why we're using negative values however, I could not find a reason. For backward compatibilities sake and to reduce our test scopes, I chose to keep it that way. IMO it's not a problem as long as the key is unique, being negative/positive is not a problem.

Reply:
I agree with you, I only asked to see if there were some other hidden reasons.
Thanks for the explanations.